src/components/SuggestionsList/SuggestionsList.tsx

```diff
@@ -0,0 +1,155 @@
+/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */
+import '../../yamui';
+import * as React from 'react';
+import { NestableBaseComponentProps } from '../../util/BaseComponent/props';
+import SuggestionsListItem, { SuggestionItem } from './SuggestionsListItem';
+import Spinner from '../Spinner';
+import Text, { TextSize, TextColor } from '../Text';
+import Callout from '../Callout';
+import Block, { GutterSize } from '../Block';
+import './SuggestionsList.css';
+
+export interface SuggestionsListProps extends NestableBaseComponentProps {
+  /**
+   * Flag to indicate that a search is in flight
+   * @default false
+   */
+  isLoading?: boolean;
+  /**
+   * Text to show next to the loading spinner.
+   */
+  loadingText: string;
+  /**
+   * Text to show when the results are empty for the active search.
+   */
+  noResultsText: string;
+  /**
+   * The groups of items to render.
+   */
+  groupedItems?: SuggestionItemGroupProps[];
+  /**
+   * The index of current keyboard selection.
+   */
+  selectedId?: string | number;
+  /**
+   * The active search that produced the current state.
+   */
+  searchText: string;
+  /**
+   * Called when an item in a group is selected by clicking.
+   */
+  onItemSelected(item: SuggestionItem): void;
+  /**
+   * The target that the Callout should try to position itself based on.
+   */
+  target: HTMLElement;
+}
+
+interface SuggestionsListWithResultsProps extends SuggestionsListProps {
+  groupedItems: SuggestionItemGroupProps[];
+}
+
+export interface SuggestionItemGroupProps {
+  title: string;
+  items: SuggestionItem[];
+}
+
+export interface SuggestionsListState {
+  hoveredId: string | number | null;
+}
+
+const hasResults = (props: SuggestionsListProps): props is SuggestionsListWithResultsProps => {
+  return !!props.groupedItems && props.groupedItems.length > 0;
+};
+
+const withResultsClass = 'y-suggestions-list--with-results';
+const withStatusClass = 'y-suggestions-list--with-status';
+
+/**
+ * A `SuggestionsList` displays a list of search results in a dropdown.
+ */
+export default class SuggestionsList
+  extends React.PureComponent<SuggestionsListProps, SuggestionsListState> {
+
+  public state: SuggestionsListState = { hoveredId: null };
```

Comment:
Why would the state be public?

Reply:
![screen shot 2017-12-14 at 9 29 34 am](https://user-images.githubusercontent.com/692240/34005938-6b90c072-e0b1-11e7-82f1-f84346144764.png)

Reply:
looks like PureComponent has it public

Reply:
Umm, I don't like seeing a `public` there, and also having to repeat the type of state.

Anything against constructor?

Reply:
nah dont care